### PR TITLE
CIRC-2050 Add useStubItem parameter to allowed-service-points endpoint

### PR DESCRIPTION
**Pull request by roman-barannyk**
## Purpose
Add a non-required boolean parameter useStubItem to an endpoint GET /circulation/requests/allowed-service-points. Default value is false.
When set to true, it should change the behavior:
Instead of running circulation rules for each item of the title, circulation rules need to be executed with the patron group parameter only. All of the item-related parameters (materialTypeId, loanTypeId, locationId) should be random UUIDs.

Resolves: [CIRC-2050](https://folio-org.atlassian.net/browse/CIRC-2050)

### CODE REVIEW

File: ramls/circulation.raml [hunk 1]
```diff
@@ -343,6 +343,12 @@ resourceTypes:
               description: "Instance ID"
               pattern: "^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[1-5][a-fA-F0-9]{3}-[89abAB][a-fA-F0-9]{3}-[a-fA-F0-9]{12}$"
               required: false
+            useStubItem:
+              description: "Allows to apply circulation rules based on patron group only"
```
Comment (OleksandrVidinieiev) [inline ramls/circulation.raml:347]: When true, allows to apply...

File: src/main/java/org/folio/circulation/domain/AllowedServicePointsRequest.java [hunk 2]
```diff
@@ -19,15 +19,15 @@ public class AllowedServicePointsRequest {
   private String requesterId;
   private String instanceId;
   private String itemId;
+  private String requestId;
+  private String useStubItem;
 
   public boolean isForTitleLevelRequest() {
     return instanceId != null;
   }
-
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/domain/AllowedServicePointsRequest.java:26]: Revert?

File: src/main/java/org/folio/circulation/infrastructure/storage/requests/RequestPolicyRepository.java [hunk 3]
```diff
@@ -95,6 +96,13 @@ public CompletableFuture<Result<Map<RequestPolicy, Set<Item>>>> lookupRequestPol
       .thenCompose(r -> r.after(this::lookupRequestPolicies));
   }
 
+  public CompletableFuture<Result<RequestPolicy>> lookupRequestPolicy(User user) {
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/infrastructure/storage/requests/RequestPolicyRepository.java:99]: Add comment regarding random IDs

File: src/main/java/org/folio/circulation/resources/AllowedServicePointsResource.java [hunk 4]
```diff
@@ -70,7 +70,7 @@ private static Result<AllowedServicePointsRequest> buildRequest(RoutingContext r
 
     AllowedServicePointsRequest request = new AllowedServicePointsRequest(operation,
       queryParams.get("requesterId"), queryParams.get("instanceId"), queryParams.get("itemId"),
-      queryParams.get("requestId"));
+      queryParams.get("requestId"), queryParams.get("useStubItem"));
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/resources/AllowedServicePointsResource.java:73]: Parse to primitive boolean

**Review by OleksandrVidinieiev — APPROVED** [review]

**Review by alexanderkurash — APPROVED** [review]

**Pull request merged**